### ISSUE TITLE
Query builder - join saved questions-custom column not found

### ISSUE THREAD

**Issue by huyangGrace** [metabase/metabase#13649]
**Describe the bug**
When I try to join two previously created questions (these were created using the query builder, not by raw SQL) I end up with this error:
```Column "source.QUANTITY" not found; ```

I try to view the native SQL, then I found the SQL is wrong:
```SELECT "source"."PRODUCT_ID" AS "PRODUCT_ID", "source"."count" AS "count", "source"."rate" AS "rate", "source"."PRODUCT_ID" AS "PRODUCT_ID_2", "source"."count" AS "count_2"
FROM (SELECT (CAST("Question 2903"."count" AS float) / CASE WHEN "source"."count" = 0 THEN NULL ELSE "source"."count" END) AS "rate", "source"."PRODUCT_ID" AS "PRODUCT_ID", "Question 2903"."PRODUCT_ID" AS "PRODUCT_ID_2", "source"."PRODUCT_ID" AS "PRODUCT_ID_3", "source"."QUANTITY" AS "QUANTITY", "Question 2903"."count" AS "count", "source"."count" AS "count_2" FROM (SELECT "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", count(*) AS "count" FROM "PUBLIC"."ORDERS"
WHERE "PUBLIC"."ORDERS"."PRODUCT_ID" = 4
GROUP BY "PUBLIC"."ORDERS"."PRODUCT_ID"
ORDER BY "PUBLIC"."ORDERS"."PRODUCT_ID" ASC) "source"
LEFT JOIN (SELECT "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", count(*) AS "count" FROM "PUBLIC"."ORDERS" WHERE ("PUBLIC"."ORDERS"."PRODUCT_ID" = 4
   AND "PUBLIC"."ORDERS"."QUANTITY" > 3) GROUP BY "PUBLIC"."ORDERS"."PRODUCT_ID") "Question 2903" ON "source"."PRODUCT_ID" = "Question 2903"."PRODUCT_ID") "source"
LIMIT 1048576
 ```

**Logs**
Here is my Diagnostic Info:
``` {
  "browser-info": {
    "language": "zh-CN",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.111 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "14.0.2+12-46",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "https://java.oracle.com/",
    "java.version": "14.0.2",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "14.0.2+12-46",
    "os.name": "Mac OS X",
    "os.version": "10.15.1",
    "user.language": "zh",
    "user.timezone": "Asia/Shanghai"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "oracle"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.18-log"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.6.2"
      }
    },
    "run-mode": "dev",
    "version": {
      "tag": "v0.37.0.2",
      "hash": "43ae7dc",
      "branch": "release/v0.37.0.2",
      "date": "2020-10-30"
    },
    "settings": {
      "report-timezone": "Asia/Hong_Kong"
    }
  }
}
```

**To Reproduce**
1. Custom question > Sample Dataset > Orders
Filter "Product ID" is equal to 4
Summarize "Count of rows" and  by "Product ID", save as Question1
![1](https://user-images.githubusercontent.com/7588235/97792578-fcbe7600-1c1a-11eb-8267-399b230811d8.png)
2. Custom question > Sample Dataset > Orders
Filter "Product ID" is equal to 4 and Quantity is greater than 3
Summarize "Count of rows" and  by "Product ID", save as Question2
![2](https://user-images.githubusercontent.com/7588235/97792583-021bc080-1c1b-11eb-9aff-69af9268b1c1.png)
3. Custom question > Saved Questions > Question1
Join Question2 by "Product ID"
Create "Custom column" use Question2's Count divide Question1' Count
Fail with error:
```
 Column "source.QUANTITY" not found; 
```

![3](https://user-images.githubusercontent.com/7588235/97792584-0647de00-1c1b-11eb-86d2-d1194b5477f3.png)

![4](https://user-images.githubusercontent.com/7588235/97792585-09db6500-1c1b-11eb-8cd2-e991820bc6b1.png)

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Related to #12762 and #12839 and #12928

**Comment by nemanjaglumac** [comment]
I can completely reuse repro (steps) from #13000 and then add a few lines to reproduce this issue also.

1. Add a custom column `[Q3(joined question) -> Sum of Rating] / [Sum of Rating], save as "Foo"
2. Click visualize
3. Fails with the same error `Column "source.RATING" not found;`
![image](https://user-images.githubusercontent.com/31325167/99127356-602ba900-2608-11eb-887b-a5f67d10afd1.png)

@flamber if you approve this I'll proceed with reusing 13000 and creating a new repro for this.

**Comment by flamber** [comment]
@nemanjaglumac Several of the nested query issues are probably very related, so fixing one might fix others, but #13000 was a frontend specific issue that broke Notebook, this issue adds the extra spicy custom column. But yes, reuse when possible.

**Comment by flamber** [comment]
Was fixed in 0.38.1, but I don't know by which PR specifically.
